migrations/helpers/DatabaseTaskTemplate.js

```diff
@@ -0,0 +1,67 @@
+class DatabaseTaskTempalte {
+	constructor({
+		id,
+		_id,
+		$set,
+		set,
+		$unset,
+		unset,
+		isModified = true,
+	}) {
+		this.id = _id || id;
+		this.set = $set || set || {};
+		this.unset = $unset || unset || {};
+		this.isModified = isModified;
+	}
+
+	setLog(log) {
+		this.log = log;
+	}
+
+	getId() {
+		return { _id: this.id };
+	}
+
+	get() {
+		const req = {};
+		if (Object.keys(this.set).length > 0) {
+			req.$set = this.set;
+		}
+		if (Object.keys(this.unset).length > 0) {
+			req.$unset = this.unset;
+		}
+		return req;
+	}
+
+	pushSet(key, value) {
+		this.set[key] = value;
+	}
+
+	pushUnset(key) {
+		this.unset[key] = 1;
+	}
+
+	exec(model, operation, _log) {
+		const log = _log || this.log;
```

Comment:
this.log should be set using one single unique method (as defined above or using constructor) this is just another option to set log - is it really required to separate it from this.log too?

Reply:
At the moment you can pass the logger at the creation, or by the execution. I think all are valid ways. If we force it over the constructor or setter it will be more performant. From the logic point of view i found better the passing by execution.